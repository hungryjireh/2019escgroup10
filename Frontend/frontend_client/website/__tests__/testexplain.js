import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import Explain from "../components/Explain/Explain";
import {ExplainHeader, ExplainSectionWrapper} from "../components/Explain/Explain";

afterEach(cleanup);

it("ExplainHeader renders", () => {
    const { asFragment } = render(<ExplainHeader />);
    expect(asFragment()).toMatchSnapshot();
});

it("ExplainSectionWrapper renders", () => {
    const { asFragment } = render(<ExplainSectionWrapper />);
    expect(asFragment()).toMatchSnapshot();
});

it("ExplainHeader renders with text", () => {
    const { getByText } = render(<ExplainHeader>Hello!</ExplainHeader>);
    expect(getByText("Hello!")).toBeTruthy;
});

it("Correct Values in Explain module", () => {
    const { getByText, getByAltText } = render(<Explain />);
    expect(getByText("How Does it Work?")).toBeTruthy;
    expect(getByText("Having trouble with an API, want to try out our products, or maybe you just want to chat? Fret not, Botty is here to help!")).toBeTruthy;
    expect(getByText("Your issues not resolved by Botty? Don't worry, you can send a ticket to one of our human administrators via Botty!")).toBeTruthy;
    expect(getByText("We will get back to your ticket as efficiently and professionally as possible. It's a promise from us! Yay, problem solved!")).toBeTruthy;
});

