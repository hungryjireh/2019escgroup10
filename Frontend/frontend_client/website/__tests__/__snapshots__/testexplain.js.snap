// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplainHeader renders 1`] = `
<DocumentFragment>
  <h1
    class="sc-EHOje egSPxn"
  />
</DocumentFragment>
`;

exports[`ExplainSectionWrapper renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bZQynM ciVANN"
  />
</DocumentFragment>
`;
